app/helpers/application_helper.rb

```diff
@@ -34,4 +34,24 @@ def decorated_session
       @_decorated_session ||= SessionDecorator.new
     end
   end
+
+  def sp_session
+    session[:sp]
+  end
+
+  def loa1_signup_context?
+    sp_session && current_user && !current_user.recovery_code.present?
+  end
+
+  def loa3_context?
+    sp_session && sp_session[:loa3] && current_user.recovery_code.present?
+  end
+
+  def loa_no_js_fallback_link
+    if loa1_signup_context?
+      destroy_user_path
+    elsif loa3_context?
+      verify_session_path
+    end
```

Comment:
You are right. I must have been looking at the wrong thing. I was also probably confused by the name of the method `loa1_signup_context?` because it is inaccurate. The cancel link will point to the `destroy_user_path` regardless of LOA level. How about renaming it to `sp_signup_context?`

Also, this brings up another question: Are we only allowing users who came from an SP to cancel the sign up process? If you sign up directly without coming from an SP, this feature will not take effect. Is that intentional?